pkg/ansible/proxy/proxy_test.go

```diff
@@ -102,18 +110,31 @@ func createPod(name, namespace string, cfg *rest.Config) runtime.Object {
 			ActiveDeadlineSeconds: &three,
 		},
 	}
-	cl, err := client.New(cfg, client.Options{})
-	err = cl.Create(context.Background(), pod)
+	cl, err := newClient(cfg)
 	if err != nil {
-		return nil
+		return nil, err
+	}
+	if err := cl.Create(context.Background(), pod); err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
+func deletePod(pod runtime.Object, cfg *rest.Config) error {
```

Comment:
We're still passing the `*rest.Config` and re-creating a client each time we call `createPod()` and `deletePod()`.

My suggestion is to make the signature of the function use `client.Client` instead of `*rest.Config` (and I would suggest moving the client to the first parameter of the function signature).

Also, since we have a controller-runtime manager, we can probably use `mgr.GetClient()` in `TestHandler()` and then pass that client to both `createPod()` and `deletePod()`.

Reply:
Actually, has no need to have a new delete pod func. Please, let me know what o you think now.